src/components/schedule-page/corrupted-schedule.component.tsx

```diff
@@ -21,13 +22,31 @@ export function CorruptedScheduleComponent(): JSX.Element {
   const { past } = useSelector(
     (state: ApplicationStateModel) => state.actualState.persistentSchedule
   );
-  const isPreviousVersionAvailable = past.length > MINIMUM_UNDO_COUNT_TO_REVERT_NORMAL_SCHEDULE;
+  const { month_number: month, year } = useSelector(
+    (state: ApplicationStateModel) => state.actualState.persistentSchedule.present.schedule_info
+  );
+
+  const isCurrentNotCorruptedSchedule = useCallback(
+    (schedule: ScheduleDataModel): boolean => {
+      return (
+        !schedule.isCorrupted &&
+        schedule.schedule_info.month_number === month &&
+        schedule.schedule_info.year === year
+      );
+    },
+    [month, year]
+  );
+
+  const currentMonthPastRevisions = past.filter((schedule) =>
+    isCurrentNotCorruptedSchedule(schedule)
+  );
+
+  const isPreviousVersionAvailable =
+    currentMonthPastRevisions.length >= MINIMUM_UNDO_COUNT_TO_REVERT_NORMAL_SCHEDULE;
```

Comment:
Skoro 
```
const currentMonthPastRevisions = past.filter((schedule) =>
    isCurrentNotCorruptedSchedule(schedule)
```
To czy `MINIMUM_UNDO_COUNT_TO_REVERT_NORMAL_SCHEDULE` nie powinno wynosić 1?

Chociaż jeden nie zepsuty, żeby móc zrewertować

Reply:
I tak i nie xD
Tak, dlatego że minimum jeden niezepsuty, żeby przywrócić poprawną wersję.
Nie, ponieważ, żeby był jeden niezepsuty to w historii muszą być 2 grafiki, bo jeden to grafik, który wywołał błąd, ale nie jest oznaczony jako corrupted.
Zmienna nazywa się MINIMUM_UNDO_COUNT_TO_REVERT_NORMAL_SCHEDULE i to faktycznie wynosi 2. 
Jeden undo, żeby wycofać isCorrupted true, drugi żeby faktczynie wycofać grafik, który jest błędny. 
Podsumowując - zostawilbym tak jak jest

Reply:
Wiem
Ale jak filtrujesz, to używasz predykat który pomija te `corrupted`

Reply:
```
  const isCurrentNotCorruptedSchedule = useCallback(
    (schedule: ScheduleDataModel): boolean => {
      return (
        !schedule.isCorrupted &&
        schedule.schedule_info.month_number === month &&
        schedule.schedule_info.year === year
      );
    },
    [month, year]
  );

  const currentMonthPastRevisions = past.filter((schedule) =>
    isCurrentNotCorruptedSchedule(schedule)
```

Reply:
Tak, ale sprawdzam tylko w Past, a jeden Corrupted jest w present.